Merge gpugraph to develop

lxsbupt opened the pull request:
<!-- Demo: https://github.com/PaddlePaddle/Paddle/pull/24810 -->
### PR types
<!-- One of [ New features | Bug fixes | Function optimization | Performance optimization | Breaking changes | Others ] -->
New features
### PR changes
<!-- One of [ OPs | APIs | Docs | Others ] -->
Others
### Describe
<!-- Describe what this PR does -->
gpugraph  二期合入develop，主要升级点
1. 新增embedding和graph图属性的多级存储，通过ssd&内存，提升参数和图属性的存储规模。
2. 新增采样、参数拉取和训练的三级流水线，最大化资源效率，提升框架性能。
3. 新增graphsage&erniesage采样，扩展图采样算法，优化策略效果。
4. 通过显存池、变长slot-feature存储等feature，优化显存利用，降低显存开销。
5. 升级多处性能损耗点，比如topo_aware_、gpu table冲突统计和跨卡直接访问等。
6. 升级cpu graph的的加载粒度和存储周期，解决大图场景的内存瓶颈。

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/PaddlePaddle/Paddle?pullRequest=48507) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/PaddlePaddle/Paddle?pullRequest=48507) before we can accept your contribution.<br/>**3** out of **9** committers have signed the CLA.<br/><br/>:white_check_mark: danleifeng<br/>:white_check_mark: Thunderbrook<br/>:white_check_mark: DesmonDay<br/>:x: chao9527<br/>:x: miaoli06<br/>:x: root<br/>:x: lxsbupt<br/>:x: qingshui<br/>:x: huwei02<br/><hr/>**root** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/PaddlePaddle/Paddle?pullRequest=48507) it.</sub>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/PaddlePaddle/Paddle?pullRequest=48507) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/PaddlePaddle/Paddle?pullRequest=48507) before we can accept your contribution.<br/>**3** out of **9** committers have signed the CLA.<br/><br/>:white_check_mark: danleifeng<br/>:white_check_mark: Thunderbrook<br/>:white_check_mark: DesmonDay<br/>:x: chao9527<br/>:x: miaoli06<br/>:x: root<br/>:x: huwei02<br/>:x: qingshui<br/>:x: lxsbupt<br/><hr/>**root** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/PaddlePaddle/Paddle?pullRequest=48507) it.</sub>

lxsbupt committed:
merge gpugraph to develop, fix code style

wuhuachaocoding committed:
update for untrainable params for stage3. (#48577)

lxsbupt committed:
merge gpugraph to develop, trigger ci

SigureMo committed:
[CodeStyle][isort][Dy2St] sort imports in test_error (#48746)

* [CodeStyle][isort][Dy2St] sort imports in test_error

* update lineno

zyfncg committed:
Clear extra input (Bias, ResidualData) in OpMaker of conv2d (#47579)

* delete Bias and ResidualData in OpMaker of conv2d

* delete extra input of conv3d

* refactor pass of conv_bias_fusion

* fix mkldnn dependency

* fix mkldnn compile

* fix test_conv_bias_mkldnn_fuse_pass

* police some code

* remove useless log

* fix analyzer_vit_ocr_tester

* fix conv_activation_mkldnn_fuse_pass

* fix test_analyzer_ocr

* add fused_conv_sig

* fix performence regression

* fix performance regression

2742195759 committed:
make bilinear interpolate stable. (#48644)

* make bilinear interpolate stable.

* fix code

ceci3 committed:
clear tmp var in ptq (#48660)

lxsbupt committed:
merge gpugraph to develop, fix py-api comment

lxsbupt committed:
merge gpugraph to develop, fix mac-python3

lxsbupt committed:
merge gpugraph to develop, fix mac-python3

SigureMo committed:
[Dy2St] replace deprecated `load_module` with `exec_module` (#48679)

lxsbupt committed:
merge gpugraph to develop, fix mac-python3

zhangyk0314 committed:
modify d2d copy to xpu::copy in xpu kernel, test=kunlun (#48710)

veyron95 committed:
rm _test_eager_guard (#48767)

201716010711 committed:
delete sampling_id api (#48543)

qili93 committed:
[NPU] add FLAGS_npu_storage_format env to enable npu storage format, test=develop (#48774)

zhoutianzi666 committed:
optimize nchw<->nhwc kernel in fp16 model (#48692)

Commit:
fix: oss just support sm>=75 (#48731)

QingshuChen committed:
update kl1 op list and optimize matmul unitest for kunlun (#48775)

*test=kunlun

0x45f committed:
Fix accuracy fp16 kernel return fp32 tensor error (#48803)

Liyulingyue committed:
[phi::DenseTensor] Replace Tensor with phi::DenseTensor (#48682)

Courtesy-Xs committed:
[Zero-Dim] Support 0D for paddle.diagflat (#48735)

* [Zero-Dim] Support 0D for paddle.diagflat

xiaoguoguo626807 committed:
【fluid api clear】Move batch norm1 (#47965)

* modify slice infershape

* code style

* modify slice_unittest

* temp fix

* batch_norm api move

* code_style

* codestyle

* ci_static

* add __init__

* reset other change

* revert .cc

* add import batchnorm

* conflict and revert

* fix bug

* fix third conflict one day

* fix conflict

* fix conflict bug

* fix conflict bug

* modify api

* code_style

* modify doc

* add lost doc stable

* fix conflict bug

* ci lack of gpu

wangzhen38 committed:
[remove fluid] PRelu BilinearTensorProduct Conv2DTranspose SequenceConv RowConv (#48654)

* [remove fluid] PRelu BilinearTensorProduct

* [remove fluid] PRelu BilinearTensorProduct Conv2DTranspose SequenceConv RowConv

* [remove fluid] PRelu BilinearTensorProduct Conv2DTranspose SequenceConv RowConv

* [remove fluid] PRelu BilinearTensorProduct Conv2DTranspose SequenceConv RowConv

* [remove fluid] PRelu BilinearTensorProduct Conv2DTranspose SequenceConv RowConv

* [remove fluid] PRelu BilinearTensorProduct Conv2DTranspose SequenceConv RowConv

* [remove fluid] PRelu BilinearTensorProduct Conv2DTranspose SequenceConv RowConv

* [remove fluid] PRelu BilinearTensorProduct Conv2DTranspose SequenceConv RowConv

lxsbupt committed:
merge gpugraph to develop, rollback graph_send_recv

zhwesky2010 committed:
fix ci (#48730)

kevinng77 committed:
Remove reduntant numpy output in Example code (1/3), test=document_fix (#48678)

Atlantisming committed:
修改了英文API文档 (#48219)

* 修改paddle.nn.dynamic_decode，paddle.nn.functional.diag_embed 示例

* mma qk tensor_core (#48087)

* use mma for QK dot computing in fused_multi_transformer.
* Update fused_multi_transformer_op.cu.h

* remove lrn which is not used in paddle 2.0 (#47945)

* replace scatter_nd and scatter_nd_add with paddle.scatter_nd and (#47960)

paddle.scatter_nd_add

* [PHI] Migrate mul_grad kernel (#48061)

* cleanup unused code

* unify is_int8 is_bfloat16

* Simplify matmul_v2 FWD kernel

* remove RunKernel methods

* remove import namespace

* remove headers

* clean fluid/phi cross imports

* remove fluid axpy_handler

* delete fluid methods

* activations

* OneDNNMemDesc

* MKLDNNFormatForSize

* MatchShapeToLayout

* MKLDNNMemoryFormat

* MKLDNNFormat

* ReorderMKLDNNHandler

* to_void_cast

* review suggestions

* interpolate

* remove fluid depedency

* init

* ExecuteMatMulV2

* rm fluid kernel

* matmul_grad

* remove mutable_data

* mul_grad

* delete unnecessary shape and slice op (#48112)

* 修改英文文档。

* 修改segment operator等英文文档。

* 重新修改了paddle.einsum，paddle.unique_consecutive，
paddle.disable_signal_handler的英文文档格式。

* 重新修改了英文文档格式。;test=docs_preview

* Update extension.py

* 重新修改了英文文档格式。;test=docs_preview

* 重新修改了英文文档格式。
待验收：
- paddle.linalg.svd
- paddle.nn.functional.diag_embed
- paddle.set_grad_enabled
- paddle.disable_signal_handler
- paddle.cumprod
- paddle.devaice.cuda.stream_guard

待修改：
- paddle.nn.dynamic_decode
- paddle.einsum
- paddle.unique_consecutive
- paddle.linalg.svd
- paddle.uncubate.segment_min
- paddle.uncubate.segment_max
- paddle.uncubate.segment_sum
- paddle.uncubate.segment_mean

;test=docs_preview

* 重新修改了英文文档格式。
待验收：
- paddle.linalg.svd
- paddle.nn.functional.diag_embed
- paddle.set_grad_enabled
- paddle.disable_signal_handler
- paddle.cumprod
- paddle.devaice.cuda.stream_guard
- paddle.nn.dynamic_decode
- paddle.unique_consecutive
- paddle.linalg.svd

待修改：
- paddle.einsum
- paddle.incubate.segment_min
- paddle.incubate.segment_max
- paddle.incubate.segment_sum
- paddle.incubate.segment_mean

;test=docs_preview

* 重新修改了英文文档格式。
待验收：
- paddle.linalg.svd
- paddle.nn.functional.diag_embed
- paddle.set_grad_enabled
- paddle.disable_signal_handler
- paddle.cumprod
- paddle.devaice.cuda.stream_guard
- paddle.nn.dynamic_decode
- paddle.unique_consecutive
- paddle.linalg.svd

待修改：
- paddle.einsum
- paddle.incubate.segment_min
- paddle.incubate.segment_max
- paddle.incubate.segment_sum
- paddle.incubate.segment_mean

;test=docs_preview

* update

* test=docs_preview

* update formula; test=docs_preview

* update formula; test=docs_preview

* remove this operator; test=docs_preview

* add hyper link; test=docs_preview

* add default value; test=docs_preview

* update format; test=docs_preview

* empty commit; test=docs_preview

* fix codestyle issues; test=docs_preview

* empty commit; test=docs_preview

Co-authored-by: lzy <569782149@qq.com>
Co-authored-by: Vvsmile <450864116@qq.com>
Co-authored-by: Sławomir Siwek <slawomir.siwek@intel.com>
Co-authored-by: RichardWooSJTU <37864677+RichardWooSJTU@users.noreply.github.com>
Co-authored-by: Ligoml <39876205+Ligoml@users.noreply.github.com>
Co-authored-by: Nyakku Shigure <sigure.qaq@gmail.com>

Silv3S committed:
[PHI] Migrate squeeze and squeeze_grad kernels (#48634)

* squeeze kernel

* squeze fwd

* whitespace

Fancy-hjyp committed:
修复paddle.nn.functinal包和paddle.nn包下API文档 (#48581)

VigiZhang committed:
assign cve number to pdsa, test=document_fix (#48846)

zhengqiwen1997 committed:
[fluid remove]: remove paddle.fluid.layers.yolo_box and paddle.fluid.layers.yolov3_loss (#48722)

* remove paddle.fluid.layers.nn.temporal_shift

* code check

* rm unittest

* remove fluid.yolo_box

* remove fluid.yolov3_loss

* change the comments of yolov3_loss to yolo_loss

lxsbupt committed:
merge gpugraph to develop, fix windows compile

lxsbupt committed:
merge gpugraph to develop, fix windows compile

lxsbupt committed:
merge gpugraph to develop, fix windows compile

Xreki committed:
Try add eval() to speedup the eigen performance. (#48855)

risemeup1 committed:
[Fluid Clean]move inplace_apis_indygraph_only from paddle.flud.dygraph.inplace_utils to paddle.utils (#48744)

* move inplace_apis_indygraph_only from paddle.flud.dygraph.inplace_utils to paddle.utils

* modify conflict

* modify conflict

* modify conflict

* modify conflict

* modify conflict

* modify conflict

* modify conflict

* modify static-check ci error

* fix conflict

* modify failed tests

* fix conflict

* fix conflict

* fix pool2d examples

* modify conflict

* fix failed tests

* fix conflict

* fix failed tests

* modfiy problem of deleting pool2d

lxsbupt committed:
merge gpugraph to develop, fix windows compile

201716010711 committed:
clean fluid task: transfer gaussian random api (#48529)

yghstill committed:
Delete duplicate quant nodes in QAT (#48751)

yjjiang11 committed:
rm autograd func dynamic eager tests (#48788)

risemeup1 committed:
Setuptools optimization (#48770)

* optimize setup.py

* modify setup.py

* modify setup.py

* modify setup.py

* modify setup.py after zhangbo reviewed

SigureMo committed:
[CodeStyle][F811] fix some test cases shadowed by the same name (#48745)

* [CodeStyle][F811] fix some unittests

* fix setup.py

* remove ignore from flake8 config

* remove repeat TestAbsDoubleGradCheck

* fix rrelu test

* fix fft ut

* add noqa in fluid.lstm ut

* add rtol and atol in test_matmul_v2_op

* update rtol

* empty commit

* empty commit

* revert changes in matmul ut and add noqa

* rename test case name

wanghuancoder committed:
set free_when_no_cache_hit default value to true (#48815)

sljlp committed:
[Clean Fluid] Rm and mv some fluid dygrah apis (#48576)

Remove fluid dygrah apis
GroupNorm
TreeConv
Move fluid dygraph apis
Flatten
SpectralNorm

jiweibo committed:
[Inference] inference add cinn interface (#48741)

GhostScreaming committed:
Clean and migrate fluid APIs of paddle.fluid.layers.control_flow (#48233)

* Merge branch 'reduce_sum' of https://github.com/GhostScreaming/Paddle into mine_fluid_clean_common.

* Fix some bugs.

* Clean APIs in python/paddle/fluid/layers/control_flow.py

* Polish code style.

* Change API.

* Fix some bugs.

* Fix some bugs.

Patrick-Star125 committed:
remove gpu_info.h from phi dependencies (#48811)

zrr1999 committed:
[Paddle Inference] Add add onehot trt converter (#48655)

* add onehot trt converter

* add unitest

* fix bug

* opt code

* fix bug

* fix depth_tensor

* fix unitest

* fix bug

* fix unitest

* fix bug

* fix bug

* fix bug

* fix bug

Patrick-Star125 committed:
[PHI decoupling] remove  bbox_util.h from phi dependencies (#48761)

* remove bbox_util.h from phi

* add file bbox_util.h

* reframe bbox_util.h

201716010711 committed:
Optimize Paddle diagonal (#47904)

Aurelius84 committed:
[API Clean]Clean __all__ to avoid exposing usless API (#48713)

* [API Clean]Clean __all__ to avoid exposing usless API

* fix import

* fix typo

* remove tracedLayer unittest

GhostScreaming committed:
Clean fluid APIs in distributed and fleet files (#48851)

* Fix bug of reduce_sum op. When input.numel() > INT32_MAX, its result
is wrong.

* Remove climits.

* Clean fluid API in paddle/distributed and paddle/fleetx folders.
Include following files:
python/paddle/distributed/__init__.py
python/paddle/distributed/collective.py
python/paddle/distributed/fleet/utils/fs.py
python/paddle/distributed/fleet/utils/hybrid_parallel_inference.py
python/paddle/distributed/fleet/utils/hybrid_parallel_util.py
python/paddle/distributed/fleet/utils/internal_storage.py
python/paddle/distributed/launch/context/device.py
python/paddle/distributed/parallel.py
python/paddle/distributed/parallel_with_gloo.py
python/paddle/distributed/spawn.py
python/paddle/framework/__init__.py
To be mentioned, 'paddle.fluid.dygraph.parallel.ParallelEnv'
 and 'fluid.framework.core' keeps unchanged in those files.
ParallelEnv is used by paddle.fluid.dygraph.parallel.DataParallel.
However, APIs in paddle.fluid.dygraph.parallel can't be
migrated to paddle.distributed, as there exists cyclic import
dependencies in modules like paddle.static, paddle.tensor. And
'fluid.framework.core' will be changed to import framework.core
after fluid.core is transmitted.

* Change TODO authors.

QingshuChen committed:
rm kunlun xpu2_op_list (#48826)

*test=kunlun

zhengqiwen1997 committed:
remove detection_output, iou_similarity and bipartite_match (#48773)

From00 committed:
Set WaiterType of kGpuSync to kCPU (#48758)

FrostML committed:
[Migrate Fluid] Migrate Decoder, BeamSearchDecoder (#48754)

jiweibo committed:
[Inference] Enable infer shape cache. (#48312)

heyanru01 committed:
[Fluid Clean] remove unfold, deformable_roi_pooling, shard_index, hard_swish, mish, uniform_random, unbind (#48451)

tianshuo78520a committed:
fix-gpups setup.py (#48888)

* fix-gpups

* test=document_fix

huangjiyi committed:
[PHI decoupling] move cuda_graph from fluid to phi (#48686)

* move cuda_graph from fluid to phi

* move device_memory_aligment from fluid to phi

* Revert "move device_memory_aligment from fluid to phi"

This reverts commit b92fcd39a0a50fdac13278f49be0237a85f3a13f.

* update xpu cmake

enkilee committed:
fix english docs typo errors (#48599)

* fix english docs typo errors

the errors in docs as same as chinese pr 5468

* update docs; test=docs_preview

Co-authored-by: Ligoml <39876205+Ligoml@users.noreply.github.com>

houj04 committed:
[XPU] add load op into oplist. (#48860)

* [XPU] add load op into oplist.

* remove test_sampling_id_op_xpu.py

lugimzzz committed:
【fluid clean】remove fluid.dygraph.rnn.lstmcell and fluid.dygraph.rnn.grucell (#48719)

FrostML committed:
refine bsd doc (#48882)

Wangzheee committed:
[Paddle Inference] General optimization for no_varlen embedding layernorm (#48580)

* general optimization no_varlen embedding layernorm

sneaxiy committed:
fix tmp directories (#48863)

yjjiang11 committed:
rm dygraph_to_static eager guard tests part2 minst2ptb_lm (#48793)

* rm dygraph_to_static eager guard tests part2 minst2ptb_lm

lxsbupt committed:
merge gpugraph to develop, fix the_one_ps.py for gpups

wangzhen38 committed:
[remove fluid] under unittesets of linear api (#48564)

* [remove fluid] under unittesets of linear api

* [remove fluid] under unittesets of linear api

* [remove fluid] under unittesets of linear api

* [remove fluid] under unittesets of linear api

* [remove fluid] under unittesets of linear api

* [remove fluid] under unittesets of linear api

* [remove fluid] fluid dygrapn linear api

* [remove fluid] fluid dygrapn linear api

* [remove fluid] fluid dygrapn linear api

kangguangli committed:
[remove fluid.layers.cross_entropy] remove unit tests (part 1) (#48726)

* replace layers.cross_entropy with paddle.entropy

* fix args

* fix codestyle

jakpiase committed:
proper fix (#48360)

Reenabled ext_reorder recording for TransDataLayoutFromOneDNN

kangguangli committed:
[remove fluid.layers.matmul] remove fluid.layers.matmul in example code (#48818)

* replace fluid.layers.matmul in fluid/io.py

* fix doc error in fluid.layers.nn.sampling_id

kangguangli committed:
remove test_auto_search_dist_matmul_op.py (#48794)

201716010711 committed:
delete mean api (#48764)

jiahy0825 committed:
clean test_op_name_conflict (#48704)

jiahy0825 committed:
opt kernel_selection error msg (#48864)

RichardWooSJTU committed:
rewrite delete_weight_dequant_linear_op_encoder/decoder pass (#48650)

* rewrite delete_weight_deqquant_linear_op_encoder/decoder pass

HarperCy committed:
[XPU] add set_value and set_value_grad (#48845)

lxsbupt committed:
merge gpugraph to develop, fix gpups ut

RachelXu7 committed:
Add QuantizedMatmul in QAT (#47997)

USTCKAY committed:
fix 'BlasAXPBY unimplemented' error with custom device (#48762)

* fix 'BlasAXPBY unimplemented' error with custom device

* fix utils CmakeLists bug

JamesLim-sy committed:
first commit (#38143)

CjhHa1 committed:
[Auto Parallel] Add cluster partition and dm to pm (#48320)

* add cluster_partition and device_meshes to process_meshes funcs

* add unitest

thisjiang committed:
fix paddle2cinn float16 type support bug (#48249)

ccrrong committed:
remove pool2d from fluid (#48512)

* remove pool2d

zhengqiwen1997 committed:
[fluid remove]: remove paddle.fluid.layers.detection_map, paddle.fluid.metrics.DetectionMAP and paddle.fluid.evaluator.DetectionMAP (#48674)

* remove paddle.fluid.layers.nn.temporal_shift

* code check

* rm unittest

* remove paddle.fluid.layers.detection_map and the class:DetectionMAP

AndPuQing committed:
[PHI decoupling] move "flags.h" from fluid to phi (#48696)

lxsbupt committed:
Merge remote-tracking branch 'develop/develop' into merge_gpugraph

lxsbupt committed:
Merge remote-tracking branch 'develop/develop' into merge_gpugraph

wuhuachaocoding committed:
add set_lr & get_lr for stage2 optimizer. (#48857)

zhiqiu committed:
move share_buffer kernel to phi (#48858)

* move share_buffer kernel to phi

* fix ut

* add source file

* fix window links

jiahy0825 committed:
[Kernel Selection] Simplify kernel selection process in phi, reduce search number to half (#47771)

* simplify SelectKernelOrThrowError function in phi

* opt kernel_selection process

* polish code, fix backend error

zyfncg committed:
Support static graph code-gen for scalar and int_array (#48792)

* add suppport_tensor for code_gen to static graph

* support code-gen for int_array

* polish code

* fix bug of data_type

jiahy0825 committed:
clean unittest test_model_cast_to_bf16 (#48705)

yjjiang11 committed:
rm dy2static eager tests part1 bert2loop (#48790)

yjjiang11 committed:
rm dygraph_to_static eager guard tests part3 reinforce2yolo (#48795)

yjjiang11 committed:
rm distribution uniform eager guard test (#48768)

* rm distribution uniform eager guard test

* review

kangguangli committed:
replace cross_entropy in python/paddle/fluid/tests/unittests/test_[a-n]*.py except test_dist_transpiler.py (#48913)

kangguangli committed:
replace cross_entropy except in python/paddle/fluid/tests/unittests/*.py && unittests/*/*.py (#48922)

zhoutianzi666 committed:
[Paddle Inference]add cutlass act set in conv_elementwise_add_act_fuse_pass (#48838)

* add cutlass act set in conv_elementwise_add_act_fuse_pass

cyber-pioneer committed:
move fluid.layers.create_global_var to static.create_global_var (#48777)

AnnaTrainingG committed:
Modified the Kernel policy. When the compute is NHWC (#48563)

HarperCy committed:
temporally disable set_value (#48942)

XiaociZhang committed:
xpu support inplace flatten (#48909)

This is a PR to catch up with latest xpu white list strategy
(https://github.com/PaddlePaddle/Paddle/pull/48606)
, since original list only include 'fluid' fashion names, but new list
must include 'phi' fashion as well.
Refer to paddle/phi/core/kernel_factory.cc for more details.

Commit:
fix:vit_attention ut (#48884)

veyron95 committed:
mv fused_bias_dropout_residual_ln to fluid manual dir (#48824)

* mv fused_bias_dropout_residual_ln to fluid manual dir

* rm useless comments

b3602sss committed:
bug fix (#48829)

huangjiyi committed:
move ops_extra_info_gen.py from phi to fluid (#48926)

MARD1NO committed:
fix scale type in alpha and beta (#48887)

zhangjun committed:
[inference][trt] upgrade prelu op  (#48528)

* add prelu

yjphhw committed:
对多个文档按照要求修改 对应中文的#5453 (#48886)

* fix doc

* test=document_fix

Co-authored-by: Ligoml <39876205+Ligoml@users.noreply.github.com>

kangguangli committed:
replace cross_entropy in python/paddle/fluid/tests/unittests/*.py except test*.py (#48919)

FrostML committed:
[remove fluid] Remove fluid APIs (#48641)

SigureMo committed:
[CodeStyle] fix renamed files not being monitored by Codestyle Check (#48892)

zhengqiwen1997 committed:
[fluid remove]: remove paddle.fluid.layers.box_coder and paddle.fluid.layers.polygon_box_transform (#48896)

* remove fluid_box_coder and polygon_box_transform

* code check

jiahy0825 committed:
[Custom XPU Support] Custom extension support xpu backend (#48733)

* support custom_xpu

* update cmake to test xpu

* support custom_xpu, verify mechanism

* fix test_custom_relu_op_xpu_setup.py, test=kunlun

* fix FLAGS_init_allocated_mem

* cancel TIMEOUT property

* reset FLAGS_init_allocated_mem property

yjjiang11 committed:
rm mlu ops eager guard tests (#48769)

yjjiang11 committed:
rm npu instance_np op for eager guard tests (#48785)

yjjiang11 committed:
remove xpu eager guard tests (#48786)

kangguangli committed:
[remove fluid.layers.cross_entropy] remove unit tests (part 3)  (#48918)

* replace cross_entropy in python/paddle/fluid/tests/unittests/test_[o-z]*.py plus test_dist_transpiler.py

* fix test_prune

yuanlehome committed:
[Inference] optimize some code and fix some bug (#48780)

* clean ir_pass_manager and fix map_depthwise_conv_to_conv_pass

* fix unitest timeout

Silv3S committed:
[PHI] Migrate reshape kernel (#48749)

* reshape

* typo

* remove header

risemeup1 committed:
support py3 in setup.py (#48905)

* support py3 in setup.py

* support setup.py bdist_wheel in py3

* support py3 in setup.py

* modify run_setup

zhoutianzi666 committed:
[Paddle-TRT] add cast between  int64 tensor  and Paddle-TRT (#45547)

* Add cast between int64 tensor and Paddle-TRT
* Add Unit testing.

sneaxiy committed:
fix sharding_stage1 amp O2 decorate bug (#48960)

wangzhen38 committed:
[remove fluid] fluid dygraph Embedding (#48806)

* [remove fluid] fluid dygraph Embedding

* [remove fluid] fluid dygraph Embedding

* [remove fluid] fluid dygraph Embedding

* [remove fluid] fluid dygraph Embedding

* [remove fluid] fluid dygraph Embedding

* [remove fluid] fluid dygraph Embedding

jiweibo committed:
fix for mkldnn (#48852)

JamesLim-sy committed:
H2D data transfer optimization with usage of structure type for stack kernel (#48899)

* first commit.

* refine performance with fast_divmod

* refine performance with fast_divmod

sljlp committed:
rm accuracy and auc in extra __all__ (#48986)

HermitSun committed:
Add dynamic checks for collective communication on NCCL  (#48915)

* chore: unify `SingleTensor`

* feat: dynamic check

sljlp committed:
support sharding in fp16 on xpu,  (#48897)

* support sharding in fp16 on xpu, change reduce_max to reduce_sum for found nan or inf

* update

From00 committed:
Support cross-step stream synchronization for standalone executor (#48809)

* Add UT

* Support cross-step stream synchronization for standalone executor

* Fix typos

* Fix typos

* Update UTs

heavyrain-lzy committed:
Generate static graph code of some ops by yaml (#48771)

* generate static graph code of some ops by yaml, test = develop

* fix 'take_along_axis' yaml style

* reset scatter/scatter_nd_add

* delete the comments of put_along_axis

zhoutianzi666 committed:
fix a bug in GetTrtWeight (#48993)

zyfncg committed:
add static_ops.yaml for static op (#48991)

huangjiyi committed:
[PHI decoupling] move norm_utils.cu.h from fluid to phi and remove norm_utils.h in fluid (#48930)

* move norm_utils.cu.h from fluid to phi

* remove norm_utils.h in fluid

* fix bugs and replace mutable_data with Alloc

* replace mutable_data with Alloc

zhoutianzi666 committed:
forbid conv op whose weight is not a persistable weight into Paddle-TRT (#48763)

Commit:
fix: Move the pass location to the appropriate location (#48951)

Xreki committed:
Enhance check_nan_inf implementation for CPU. (#48591)

* Enable to print device info.

* Enhance the nan and inf checking for cpu.

* Implement a common print function.

* Unify the check of complex numbers.

* Rewrite the omp method.

* Count and print the number of nan and inf.

* Change the print content.

* Add unittest.

Commit:
[PHI] OneDNN version of Copy (#48539)

* OneDNN version of Copy, tranpose kernels adjusted

* style fixes in tranpose_grad

* redundant headers deleted

Commit:
fix: there are some bugs with trt 8.0 (#48921)

* fix: there are some bugs with trt 8.0

* fix:windows CI trt is too old

Courtesy-Xs committed:
Optimization of Eigh op with ssyevj_batched runtime api (#48560)

* fix codestyle

* add double complex<float> complex<double> dtype support for syevj_batched

* fix use_syevj flag for precision loss when input dtype of syevj_batch is complex128 in some case

* optimize eigh in different case

* fix missing ; bug

* fix use_syevj bug

* fix use_cusolver_syevj_batched flag

kangguangli committed:
replace cross_entropy in python/paddle/fluid/tests/unittests/*/*.py except unittests/*.py (#48920)

Patrick-Star125 committed:
[PHI decoupling] replace dependency of inclusive_scan.h from phi (#48980)

* replace dependency of inclusive_scan.h from phi

* format code

feifei-111 committed:
fluid API magration : Assert, increment, cond (#48885)

jiahy0825 committed:
[Clean fluid] Add inner function _elementwise_op_with_axis (#48748)

* add inner function _elementwise_op_with_axis

* fix transformer_model

* polish API code

* remove elementwise_div/mul api

* delete API in __all__

* delete elementwise_mul completely

* polish elementwise_mul call

* polish internal api

* resolve conflict, fix rnn.py

* use non-inplace call

* delete elementwise_mul api test

* delete elementwise_mul api test

* clean elementwise_add/sub

* restore _elementwise_op_in_dygraph in nn.py

jiweibo committed:
test_convert_to_mixed_precision.py use tempfile for temporary models/params (#48819)

YuanRisheng committed:
Tighten the Interception strategy (#48947)

* test approve ,test=document_fix

* test approve ,test=document_fix

* test approve ,test=document_fix

SigureMo committed:
[CodeStyle][isort][F401] fix some regression issues (#48936)

* [CodeStyle][isort][F401] fix some regression issues

* add import paddle to fix eval call

yjjiang11 committed:
rm multinode eager guard tests (#48766)

* rm multinode eager guard tests

* remove unwanted tests

* reset process_mpi test

yjjiang11 committed:
rm unittests eager guard tests part5 dataloader2dygraph_mnist (#48816)

YuanRisheng committed:
[PHI]Add new Tensor type and migrate save_combine kernel (#47856)

* add new tensor

* fix windows compile bugs

* fix ci bugs

* fix ci bugs

* fix ci bugs

* perfect according comment

* fix ci compile bugs

* add raw tensor

* fix ci bugs

* modify code by comment

* delete String

risemeup1 committed:
[Fluid Clean]move BatchNorm from flud.dygraph.nn to paddle.nn.layer.norm (#48734)

* move BatchNorm from flud.dygraph.nn to paddle.nn.layer.norm

* modfiy conflict

* modify pre-commit error

* modify static-check ci error

* fix failed tests

* modify conflict

* modify conflict

* delete import modelu GRUUnit

* fix falied test

* fix failed testes

* fix failed tests

* fix failed tests

* fix failed test

* fix error in test_fused_resenet_basic_block_op_xpu.py

* modify after xiaoguang reviewed

Aurelius84 committed:
[Setup] Ignore @PADDLE_BINARY_DIR@ files (#49002)

* [Setup] Ignore @PADDLE_BINARY_DIR@ files

* test=document_fix

jczaja committed:
reshape onednn test reimplemented (#48850)

* - UT reshape onednn

- Fix

test

test2

- test4

- test5

- test6

test7

- test8

- Ut reinvented

- cosmetic

* - fix

* - fix

* - fix

* - fix

* - Fix

* - fix

* - fix

* - fix

* - Fix

* lint

RichardWooSJTU committed:
update fused_multi_transformer_encoder_pass support GPT new matmul API (#48953)

* fit paddle.matmul in fleetx.gpt

wanghuancoder committed:
Revert "set free_when_no_cache_hit default value to true (#48815)" (#48968)

This reverts commit 592ed40b58d7bc015de87368cc611e865bdcd6ea.

Wangzheee committed:
[Paddle Inference]fix some transformer unitest (#48929)

* fix some transformer unitest

weishengying committed:
Enable Generic-Plugin support FP16 (#48807)

yghstill committed:
support conv1d quant & skip calibrate zero-size tensor (#48912)

engineer1109 committed:
enable custom device save model on device memory && fix conflict (#48221)

wangzhen38 committed:
[api move] cvm (#48989)

* [api move] cvm

* [api move] cvm

* [api move] cvm

* [api move] cvm

* [api move] cvm

* [api move] cvm

* [api move] cvm

* [api move] ci test

* [api move] ci test

* [api move] ci test

XiaociZhang committed:
Bugfix: xpu now only support single node multi-card, bkcl_comm_num should always set to 1 (#48961)

lxsbupt committed:
Merge remote-tracking branch 'develop/develop' into merge_gpugraph

lxsbupt committed:
Merge remote-tracking branch 'develop/develop' into merge_gpugraph

yjjiang11 committed:
rm unittests eager guard tests part23 where2zeros (#48895)

yjjiang11 committed:
rm unittests eager guard tests part17 number2pool1d (#48840)

qili93 committed:
[NPU] fix FLAGS_npu_storage_format flag in python, test=develop (#48976)

yjjiang11 committed:
remove fleet eager guard tests (#48765)

yjjiang11 committed:
rm unittests eager guard tests part6 eager_run2expand_v2 (#48817)

yjjiang11 committed:
rm unittests eager guard tests part12 imperative_optimizer2resnet (#48833)

lugimzzz committed:
[fluid clean] remove 4 fluid.layers api and imigrate 2 fluid.layer api (#48972)

* fluid clean layer

* docs

kangguangli committed:
remove reset reference in unittest for `fluid.layers.cross_entropy` (#49012)

kangguangli committed:
replace cross_entropy in test*.py except python/paddle/fluid/tests/unittests/*.py (#48978)

ccrrong committed:
remove linear_chain_crf and crf_decoding from fluid (#48996)

* remove linear_chain_crf and crf_decoding

heavyrain-lzy committed:
Generate static graph code of some ops by yaml (#48977)

* generate static graph code of some ops by yaml

* fix the code-style of yaml

* fix the framework_ci for triangular_solve

* change the 'data_type' of scatter

* add the 'out: Out' of scatter_nd_add

gouzil committed:
[tools] Update summary env (#48627)

* [tools] remove deprecated api , fix macOS get version error

* [tools] Rename the value that returns null

* [tools] add gcc, clang, cmak, libc version

* [tools] fix cudnn read error

* [tools] add gpu devices list, drive based

* [issue] update 3_build-installation-issue.yml

* [tools] fix get gpu list AttributeError

DrRyanHuang committed:
[Dy2St] transforms.RandomVerticalFlip Support static mode (#49024)

* add static RandomVerticalFlip

* object => unittest.TestCase

lxsbupt committed:
Merge remote-tracking branch 'develop/develop' into merge_gpugraph

lxsbupt committed:
Merge remote-tracking branch 'develop/develop' into merge_gpugraph

sneaxiy committed:
Save fused_attention op memory when dropout_rate = 0.0 (#48902)

* save fused_attention memory when dropout_rate = 0.0

* add ut

* fix ut bug

* fix fused_layernorm_residual_dropout_bias_test.cu

Commit:
Correct multiple inputs and outputs (#48872)

SigureMo committed:
[CodeStyle][isort][Dy2St] sort imports for paddle.jit (#48637)

* isort jit

* refine comment

zoooo0820 committed:
remove non-public apis from __all__ (#48952)

* remove non-public apis from __all__

* fix code style

wanghuancoder committed:
fix rmsprop_ yaml bug (#49026)

* fix rmsprop_ yaml bug

jjyaoao committed:
Fixed the dead link bug in the API documentation (#48969)

* first pr

* Revise nn.py

* Revise nn.py 2.0

* Revise rnn.py;test=document_fix

* test=document_fix

Co-authored-by: Ligoml <39876205+Ligoml@users.noreply.github.com>

lxsbupt committed:
Merge remote-tracking branch 'develop/develop' into merge_gpugraph

lxsbupt committed:
Merge remote-tracking branch 'develop/develop' into merge_gpugraph

Xreki committed:
Change mutable_data to ctx.Alloc. (#49001)

zhangjun committed:
[inference][trt] add more unary op and square (#48534)

* add more unary op and square

risemeup1 committed:
Support ninja (#48932)

* move inplace_apis_indygraph_only from paddle.flud.dygraph.inplace_utils to paddle.utils

* modify conflict

* modify conflict

* modify conflict

* modify conflict

* modify conflict

* modify conflict

* modify conflict

* modify static-check ci error

* fix conflict

* modify failed tests

* fix conflict

* fix conflict

* fix pool2d examples

* modify conflict

* fix failed tests

* fix conflict

* fix failed tests

* modfiy problem of deleting pool2d

* support Ninja in setup.py

* support different cmake_generators

* modify after reviewed

* delete unused denotes

HulekJakub committed:
Deleted mkldnn_inplace_pass code (#47818)

* Deleted mkldnn_inplace_pass code

* Fixed error with cmake

* Resolve conflicts

tianshuo78520a committed:
hide log (#49045)

* test=doucment_fix

* test=document_fix

Commit:
[Sparse]Optimize performance of sparse conv on T4 (#49009)

zhengqiwen1997 committed:
modify cmake file for cuda11.8 compile (#49020)

* modify cmake file for cuda11.8 compile

* add op_library(fused_embedding_eltwise_layernorm_op DEPS bert_encoder_functor)

ccrrong committed:
remove dropout from fluid (#48319)

* remove dropout

lxsbupt committed:
Merge remote-tracking branch 'develop/develop' into merge_gpugraph

lxsbupt committed:
Merge remote-tracking branch 'develop/develop' into merge_gpugraph

XiaociZhang committed:
nullptr bugfix for XPU pg mode (#49043)

* nullptr bugfix for XPU pg mode

Also a few kernels is added to xpu whitelist

* increase error msg length

JamesLim-sy committed:
Divide elementwise case from BroadcastKernel and refine transpose autotune (#33051)

* First Commit.

* add some codes

* add elementwise loader

* fix code styles

* merge with develop

* add some changes both in elementwise and transpose

* add init operation in broadcast kernel.

* change codes according to pr suggestions about transpose file

* fix error for op-benchmark ci

* fix according to ci

XieYunshen committed:
add condition of skipif (#48791)

* add condition of skipif

* fix code format error

* Update test_fused_gate_attention_op.py

update

yjjiang11 committed:
rm unittests eager guard tests part9 histogram2imperative_dataloader (#48825)

* rm unittests eager guard tests part9 histogram2imperative_dataloader

* rm basic

yjjiang11 committed:
rm unittests eager guard test part14 initializer2layer_norm (#48835)

* rm unittests eager guard test part14 initializer2layer_norm

* monior change

JZ-LIANG committed:
[Bugfix] recompute dep filter param (#49010)

* recompute dep filter param

* recompute dep for reshard

yuanlehome committed:
[Paddle Inference] rewrite convert_to_mixed_precision (#48853)

AndPuQing committed:
[CodeStyle] fix c++17-extensions warning on macos (#49017)

* fix c++17-extensions warning on macos

* fix type

fix c++17-extensions warning on macos

fix c++17-extensions warning on macos

laitingsheng committed:
Add custom CUDNN finding paths for 64bit Windows (#49066)

zhengqiwen1997 committed:
remove prior_box (#49006)

* remove prior_box

* modify the sequence of paras of prior_box in multi_box_head api

Ayuan2021 committed:
InstanceNorm1D、InstanceNorm2D、InstanceNorm3D (#48940)

* modified:   python/paddle/nn/layer/norm.py

* modified:   python/paddle/nn/layer/norm.py

* modified:   python/paddle/nn/layer/norm.py

* modified:   python/paddle/nn/layer/norm.py

* modified:   python/paddle/nn/layer/norm.py

* modified:   python/paddle/nn/layer/norm.py

* test=docs_preview

* InstanceNorm2D中文档格式修改

* test=docs_preview

* modified:   python/paddle/nn/functional/loss.py
	modified:   python/paddle/nn/functional/norm.py
	modified:   python/paddle/nn/layer/loss.py
	modified:   python/paddle/nn/layer/norm.py

* test=docs_preview

* test=docs_preview

zhaoyinglia committed:
[AutoParallel] recompute tuning (#48608)

* [AutoParallel] recompute tuning

* fix conflict

* update comment

* bug fix

* update rc algo

* tiny fix

* fix clear process_group

* remove comment

* update segment print

* fix import OpRole

* adapt amp pass and grad_clip pass for opt_tuner

* update tuning config

* fix import

* annotate recompute info on ops and upgrade recompute pass

* add op_namescope for seed op

* record reserved vars

* fix recompute var's dist_attr

* fix strategy unittest

* adapt for fp16

* update unittest

* revert copy opt

* update unittest

* rename set_recompute_segments

* fix unittest

feifei-111 committed:
fluid API magration : array_read, array_write (#49022)

* del array_write & array_read

* fix import err

* fix import err

* fix example codes

sljlp committed:
Keep double-buffer reader for static mode  (#49068)

mingxu1067 committed:
Fix nullptr to TestFuseGemmEpilogueReluBWDFP* (#48997)

wangxn12138 committed:
support fp16 index sample (#47897)

* add index sample fp16 support

* remove fluid APIs in distributed_strategy.py and role_maker.py

* Revert "remove fluid APIs in distributed_strategy.py and role_maker.py"

This reverts commit 223bbee990d3bf69e252fc3c0f19e3873550a264.

* fix instantiated more than once

* clean codes

yjjiang11 committed:
rm unittest eager guard tests part20 sparse_mv2split (#48879)

yjjiang11 committed:
rm unittests eager guard tests part11 imperative_layer2ocr (#48828)

* rm unittests eager guard tests part11 imperative_layer2ocr

* review

yjjiang11 committed:
rm eager guard tests part3_1 (#49059)

HermitSun committed:
fix: gloo compatible (#49084)

yjjiang11 committed:
rm eager guard tests part3_3 (#49061)

haohongxiang committed:
fix bug (#49081)

jiweibo committed:
[Inference] memory_optimize and mkdlnn  problem (#49054)

* memory_optimize and mkdlnn problem

* update

* update

* update

HydrogenSulfate committed:
Remove/move 16 fluid APIs (#48377)

* remove density_prior_box

* remove anchor_generator

* remove roi_perspective_transform

* remove generate_proposal_labels

* remove generate_mask_labels

* remove generate_proposals

* remove box_clip

* remove retinanet_detection_output

* remove multiclass_nms

* remove locality_aware_nms

* remove matrix_nms

* remove distribute_fpn_proposals

* remove box_decoder_and_assign

* remove collect_fpn_proposals

* remove 2 trt files

* move prior_box to static/nn/common.py

* move multi_box_head to static/nn/common.py

* fix for CI/CE

* remove retinanet_detection_output

* restore compile_vs_runtime_white_list.py

* restore test_retinanet_detection_output to white list

* replace nn.flatten by paddle.flatten, and fix doc for retinanet_target_assign

* add enable_static in demo and fix bug

* remove roi_perspective_transform in test_layers

* remove multi_box_head

* change self.multiclass_nms to _legacy_C_ops.multiclass_nms

* empty commit

* empty commit

* check code style

* fix prior_box

* fix CI

* remove redundant prior_box in detection.py

* fix docs

* remove detection

* fix prior_box en doc

* delete prior_box in common

* remote proir_box from __init__.py

Wangzheee committed:
fix embedding multihead (#49085)

From00 committed:
SetDeviceId in StreamSafeCUDAAllocation (#49080)

Silv3S committed:
[PHI decoupling] Remove fluid imports from MKLDNN code (#48981)

* fix wrong handler name

* mkldnn_engine -> onednn_engine

* remove fluid/errors.h imports

* remove fluid/enforce.h imports

* remove note and unnecessary import

* remove fluid/pretty_log.h imports

* remove fluid/place.h imports

* remove fluid/data_layout_transform.h imports

* remove fluid/device_context.h imports

* remove mkldnn_helper code

* remove fluid/mkldnn_reuse.h imports

* pretty_log import

kangguangli committed:
replace cross_entropy in python/paddle/fluid/tests/unittests/*.py (#48975)

kk-2000 committed:
修复paddle.amp.decorate等API的文档 (#48983)

* 涉及到的api有
paddle.amp.decorate
paddle.static.npu_places
paddle.signal.istft
paddle.signal.stft
paddle.linalg.eigvalsh
paddle.randint_like

* change signal.stft

* randint_like的low增加optional

* ; test=docs_preview

* 修改了注解格式; test=docs_preview

* 修改了公式格式

* 修改了decorate的models等

* test=document_fix

Co-authored-by: Ligoml <39876205+Ligoml@users.noreply.github.com>

LearningPawn committed:
按在线文档需求 61~70 更新了部分文档 (#49014)

* Update docstring:
1. 去除 python/paddle/tensor/manipulation.py 中 cast 函数描述中的 This OP；
2. 调整 python/paddle/fluid/layers/control_flow.py 中 Print 函数中参数描述的顺序，添加 optional 描述；
3. 为 python/paddle/tensor/logic.py 中 logical_and 函数添加 optional 描述；
4. 为 python/paddle/fluid/reader.py 中 DataLoader 类中 from_generator、from_dataset 函数添加 optional 描述；
5. 在 python/paddle/fluid/layers/nn.py 中 crf_decoding 函数的 param_attr 在使用中确实可视为存在默认值 None，故添加 optional 描述；
6. 修复 python/paddle/static/nn/common.py 中 data_norm 函数描述里 tex 语法错误的问题，并一并修复同一文件中的相同问题。

* 根据 review 意见修改部分内容。

* 将谓语动词去掉第三人称单数形式。

* 同步中文文档变更。

* string-->str; test=document_fix

Co-authored-by: Ligoml <39876205+Ligoml@users.noreply.github.com>

lxsbupt committed:
Merge remote-tracking branch 'develop/develop' into merge_gpugraph

lxsbupt committed:
Merge remote-tracking branch 'develop/develop' into merge_gpugraph

lxsbupt committed:
merge gpugraph to develop, fix gloo wrapper

lxsbupt committed:
merge gpugraph to develop, fix gloo wrapper

lxsbupt committed:
merge gpugraph to develop, fix ci

lxsbupt committed:
merge gpugraph to develop, fix ci

lxsbupt committed:
Merge branch 'merge_gpugraph' of https://github.com/lxsbupt/Paddle into merge_gpugraph

lxsbupt committed:
merge gpugraph to develop, fix fleet.py

lxsbupt committed:
merge gpugraph to develop, merge latest conflicts

lxsbupt committed:
merge gpugraph to develop, fix merge error

lxsbupt committed:
merge gpugraph to develop, fix merge error

lxsbupt committed:
merge gpugraph to develop, fix conflicts

lxsbupt committed:
merge gpugraph to develop, add python ut

lxsbupt committed:
merge gpugraph to develop, fix code style

lxsbupt committed:
merge gpugraph to develop, add c++ ut

lxsbupt committed:
merge gpugraph to develop, fix code style

lxsbupt committed:
merge gpugraph to develop, fix data_feed.h

lxsbupt committed:
merge gpugraph to develop, fix code style

lxsbupt committed:
merge gpugraph to develop, fix code style

lxsbupt committed:
merge gpugraph to develop, fix code style

lxsbupt committed:
merge gpugraph to develop, fix code style

luotao1 commented:
LGTM for PR-CI-Codestyle-Check, 分布式相关的C++代码格式问题 @danleifeng 后续会找其他同学修，本次先豁免
```
2022-12-21 08:21:30 paddle/fluid/distributed/ps/service/ps_local_client.h:186:  "virtual" is redundant since function is already declared as "override"  [readability/inheritance] [4]
2022-12-21 08:21:30 paddle/fluid/distributed/ps/service/ps_local_client.h:199:  "virtual" is redundant since function is already declared as "override"  [readability/inheritance] [4]
2022-12-21 08:21:30 Done processing paddle/fluid/distributed/ps/service/ps_local_client.cc
2022-12-21 08:21:30 Done processing paddle/fluid/distributed/ps/service/ps_local_client.h
2022-12-21 08:21:30 Total errors found: 30
2022-12-21 08:21:30 paddle/fluid/distributed/ps/table/ssd_sparse_table.cc:552:  Missing username in TODO; it should look like "// TODO(my_username): Stuff."  [readability/todo] [2]
2022-12-21 08:21:30 paddle/fluid/distributed/ps/table/ssd_sparse_table.cc:581:  Missing username in TODO; it should look like "// TODO(my_username): Stuff."  [readability/todo] [2]
2022-12-21 08:21:30 Done processing paddle/fluid/distributed/ps/table/ssd_sparse_table.cc
2022-12-21 08:21:30 Total errors found: 2
2022-12-21 08:21:30 paddle/fluid/framework/fleet/heter_ps/test_cpu_query.cu:31:  Static/global string variables are not permitted.  [runtime/string] [4]
2022-12-21 08:21:30 Done processing paddle/fluid/framework/fleet/heter_ps/test_cpu_query.cu
2022-12-21 08:21:30 Total errors found: 1
```

yaoxuefeng6 reviewed (approved):
LGTM

lanxianghit reviewed (approved):
LGTM for Flags

zhiqiu reviewed (approved):
LGTM for const_cast

qili93 commented:
@yaoxuefeng6 这个 PR 关于 rocksdb.cmake 的修改 broken 了 Paddle 在 aarch64 环境下的源码编译功能，辛苦修复下!
 
这里 rocksdb.cmake 里面的 sse 以及 -mpclmul 相关的编译选项都是 x86 环境下独有的，在 aarch64下会导致编译失败

![image](https://user-images.githubusercontent.com/16605440/210299299-01cfef9c-b7bf-44d9-892f-0aa93bec5a63.png)

rocksdb 本身是支持 aarch64 平台下的编译的，请根据 rocksdb 的 cmake 自动根据编译环境来选择编译选项，代码见 https://github.com/Thunderbrook/rocksdb/blob/main/CMakeLists.txt#L261

![image](https://user-images.githubusercontent.com/16605440/210299480-60a0138f-73ad-4c09-b395-3561b153dd98.png)

当前 work-around 办法是在 aarch64 编译时加上 WITH_PSCORE=OFF 来关闭对于 rocksdb 的编译和引用。
